src/Sentry.Unity.Native/SentryNative.cs

```diff
@@ -75,12 +81,37 @@ public static void Configure(SentryUnityOptions options, ISentryUnityInfo sentry
                 crashedLastRun = SentryNativeBridge.HandleCrashedLastRun(options);
                 PerDirectoryCrashInfo.Add(cacheDirectory, crashedLastRun);
 
-                options.DiagnosticLogger?
+                Logger?
                     .LogDebug("Native SDK reported: 'crashedLastRun': '{0}'", crashedLastRun);
             }
         }
         options.CrashedLastRun = () => crashedLastRun;
+
+        ShouldReinstallBackend = true;
     }
 
-    public static void ReinstallBackend() => SentryNativeBridge.ReinstallBackend();
+    // We're calling this in `BeforeSceneLoad` instead of `SubsystemRegistration` as it's too soon and the
+    // SignalHandler would still get overwritten.
+    [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+    private static void ReinstallBackend()
+    {
+        // The backend should only be reinstalled if the native SDK has been initialized successfully.
+        if (!ShouldReinstallBackend)
+        {
+            Logger?.LogDebug("Skipping reinstalling the native backend as the initialization seems to have failed.");
```

Comment:
We avoid reinstalling it if fails below so not sure I get this message.

The flag right now is flipped if native support is disabled, right? So there could be different reasons why we disabled it. 

```suggestion
            Logger?.LogWarning("Skipping reinstalling the native backend.");
```